resources/prosody-plugins/token/util.lib.lua

```diff
@@ -30,11 +135,99 @@ local function _verify_token(token, appId, appSecret, disableRoomNameConstraints
 		return nil, "'room' claim is missing";
 	end
 
+    local audClaim = claims["aud"];
+    if audClaim == nil then
+        return nil, "'aud' claim is missing";
+    end
+
 	return claims;
 end
 
-function _M.verify_token(token, appId, appSecret, disableRoomNameConstraints)
-	return _verify_token(token, appId, appSecret, disableRoomNameConstraints);
+local function _verify_token(session, token)
+
+    if token == nil then
+        if allowEmptyToken then
+            return true;
+        else
+            return false, "not-allowed", "token required";
+        end
+    end
+
+    local pubKey;
+    if asapKeyServer and session.auth_token ~= nil then
+        local dotFirst = session.auth_token:find("%.");
+        if not dotFirst then return nil, "Invalid token" end
+        local header = json.decode(basexx.from_url64(session.auth_token:sub(1,dotFirst-1)));
+        local kid = header["kid"];
+        if kid == nil then
+            return false, "not-allowed", "'kid' claim is missing";
+        end
+        pubKey = get_public_key(kid);
+        if pubKey == nil then
+            return false, "not-allowed", "could not obtain public key";
+        end
+    end
+
+    -- now verify the whole token
+    local claims, msg;
+    if asapKeyServer then
+        claims, msg = _verify_jwt_token(token, appId, pubKey, disableRoomNameConstraints);
+    else
+        claims, msg = _verify_jwt_token(token, appId, appSecret, disableRoomNameConstraints);
+    end
+    if claims ~= nil then
+        -- Binds room name to the session which is later checked on MUC join
+        session.jitsi_meet_room = claims["room"];
```

Comment:
It feels weird that method named "_verify_token" which takes 'session' as an argument magically adds some attributes to it. It should be documented with some warning at least in the method description. Or maybe the name should be different.

Also I have the impression that all methods have grown significantly and there is no documentation whatsoever to any config options, so now it's quite hard to figure out what's going on.